### ISSUE TITLE
Explosions are fucked

### ISSUE THREAD

**Issue by N3X15** [d3athrow/vgstation13#6630]
Explosions occur, but don't deliver damage to floors or objs.

# Reproduction Steps

1. Make boom

# Expected results

Boom and damage

# Observed Results

Boom, particles, no damage

**Comment by N3X15** [comment]
This happened twice in one round, from my observations: Once, when a SME melted down, and lastly, when someone threw an E20.

**Comment by N3X15** [comment]
/cc @DeityLink

Doesn't even delete the bomb.

**Comment by DeityLink** [comment]
on it

**Comment by DeityLink** [comment]
Turns out that @RemieRichards was wrong.
You can't replace 
>for(var/sstep=max_steps-1;sstep>=0;sstep--)
with
>for(var/sstep in max_steps-1 to 0)

**Comment by RemieRichards** [comment]
@DeityLink excuse me, I'm *NOT* wrong, In my testing setting X in Y to Z where Y>Z resulted in negative steps, however in your case I guess it didn't, you can force it by adding ```step -1``` onto the end of the for loop, and it IS worth it as it's faster than ```for(var/sstep=max_steps-1;sstep>=0;sstep--)```

tl;dr
```for(var/sstep in max_steps-1 to 0 step -1)``

**Comment by DeityLink** [comment]
Well seeing as even my version can instantly add every turf in Z Level to a list, I think I'll settle with that. It's not as if the proc was doing anything complicated.
